packages/common/container-definitions/api-report/container-definitions.legacy.alpha.api.md

```diff
@@ -99,13 +99,17 @@ export interface IContainer extends IEventProvider<IContainerEvents> {
     readonly audience: IAudience;
     readonly clientId?: string | undefined;
     close(error?: ICriticalContainerError): void;
+    close(error: ICriticalContainerError, disconnectReason: "Corruption"): void;
```

Comment:
looking at the change and description, i don't think these overloads are necessary to expose on the IContainer interface. I think you could limit most if not all of this to being internal and only on the implementations. this would simplify things quite a bit.

Reply:
if we want to do that i'd like to see a spec. we already have a number of ways to categorize errors such as error type, and i don't have a clear picture of how they all fit together, as they seem to have overlapping purposese.

Reply:
it also seems strange the caller of close/dispose would pick this for a customer, wouldn't they want our categorization, not one they choose?

Reply:
the more i think about this, the more i think we should just use error type. it already has a much deeper and more robust categorization that what is being added here. i'd probably also keep the server agnostic to the specific types, and have them just consumer it as a string, as i assume they just need it more telemetry, and they can do categorization at data analysis time

Reply:
+ @markfields who is our error guru

Reply:
> the more i think about this, the more i think we should just use error type. it already has a much deeper and more robust categorization that what is being added here. i'd probably also keep the server agnostic to the specific types, and have them just consume it as a string, as i assume they just need it more telemetry, and they can do categorization at data analysis time

I generally like this direction. My only counterargument is if we wanted to provide non-failure-mode disconnect reasons, but I don't know how valuable this would actually be in practice. If not valuable, then I like the error type pattern.

Reply:
that sounds like scope creep to me, and it should be a non-goal. specifically, we do not need it for the task at hand, and we have all the capabilities we need to accomplish the task at hand, so should reuse those, we shouldn't build things just because we might need them.

Reply:
I'm not making suggestion; I'm simply asking a question. If we have a longer-term desire to do something like that, we should take that into consideration so as not to paint ourselves into a corner.